xtask/src/init.rs

```diff
@@ -0,0 +1,283 @@
+use color_eyre::eyre::Result;
+use fs_extra::{dir, error, file};
+use std::io::Write;
+use std::path::PathBuf;
+use tracing::info;
+use xshell::{cmd, Shell};
+
+/// Initialize the workspace.
+pub(crate) fn init() -> Result<()> {
+    let sh = Shell::new()?;
+
+    let xtask_directory = PathBuf::from(format!("{}/target/xtask/", env!("CARGO_WORKSPACE_DIR")));
+    let wpilib_directory =
+        PathBuf::from(format!("{}/allwpilib", xtask_directory.to_str().unwrap()));
+    let ni_libraries_directory = PathBuf::from(format!(
+        "{}/ni-libraries",
+        xtask_directory.to_str().unwrap()
+    ));
+    let crate_directory = PathBuf::from(format!(
+        "{}/crates/wpilib-sys/",
+        env!("CARGO_WORKSPACE_DIR")
+    ));
+    let include_directory = PathBuf::from(format!("{}/include", crate_directory.to_str().unwrap()));
+    let lib_directory = PathBuf::from(format!("{}/lib", crate_directory.to_str().unwrap()));
+
+    // Create the above directories not exists.
+    for directory in &[&xtask_directory, &include_directory, &lib_directory] {
+        match dir::create(directory, false) {
+            Ok(_) => (),
+            Err(e) => match e.kind {
+                error::ErrorKind::AlreadyExists => (),
+                _ => panic!("Failure in creating {:?}: {:?}", directory.file_name(), e),
+            },
+        };
+
+        assert!(directory.exists())
+    }
+
+    if !wpilib_directory.exists() {
+        info!("Downloading WPILib...");
+        let wpilib_directory = wpilib_directory.to_str().unwrap();
+        cmd!(
+            sh,
+            "git clone --quiet --depth 1 --branch v2022.4.1 https://github.com/wpilibsuite/allwpilib {wpilib_directory}"
+        )
+        .ignore_stdout()
+        .ignore_stderr()
+        .run()?;
+    }
+
+    if !ni_libraries_directory.exists() {
+        info!("Downloading NI libraries...");
+        let ni_libraries_directory = ni_libraries_directory.to_str().unwrap();
+        cmd!(
+            sh,
+            "git clone --quiet --depth 1 --branch v2022.4.0 https://github.com/wpilibsuite/ni-libraries {ni_libraries_directory}"
```

Comment:
I'd like to move towards having actual *-sys crates, see #66.

Reply:
Sure, I agree, but my goal while migrating to xtask was to keep functionality as it was. This could go in a separate PR, or even better, in Arfur.